main.go

```diff
@@ -239,8 +315,49 @@ func extURL(s string) (*url.URL, error) {
 	return u, nil
 }
 
-func listen(router *route.Router) {
-	if err := http.ListenAndServe(*listenAddress, router); err != nil {
+func listen(listen string, router *route.Router) {
+	if err := http.ListenAndServe(listen, router); err != nil {
 		log.Fatal(err)
 	}
 }
+
+type stringset map[string]struct{}
+
+func (ss stringset) Set(value string) error {
+	ss[value] = struct{}{}
+	return nil
+}
+
+func (ss stringset) String() string {
+	return strings.Join(ss.slice(), ",")
+}
+
+func (ss stringset) slice() []string {
+	slice := make([]string, 0, len(ss))
+	for k := range ss {
+		slice = append(slice, k)
+	}
+	sort.Strings(slice)
+	return slice
+}
+
+func mustHardwareAddr() string {
+	ifaces, err := net.Interfaces()
```

Comment:
I think it's worth adding a sanity check for at least, I don't trust all cloud providers and network abstractions to have globally unique MAC addresses.